pandas/core/indexes/base.py

```diff
@@ -4883,8 +4883,8 @@ def get_indexer_non_unique(self, target):
             tgt_values = np.asarray(target)
         else:
             tgt_values = target._get_engine_target()
-
-        indexer, missing = self._engine.get_indexer_non_unique(tgt_values)
+        engine = self._engine
+        indexer, missing = engine.get_indexer_non_unique(tgt_values)
```

Comment:
Same as above, this line was perfectly fine

Reply:
Same here